### PR TITLE
[BE-763] Checkout.com payment using source_id

### PR DESCRIPTION
**Pull request by inai-harish**
## Description:

- Adding changes for making payments using the `source_id` created at checkout.com for recurring payments.

### Testing:

- Added tests to assert the changes - both unit and remote.

### CODE REVIEW

File: lib/active_merchant/billing/gateways/checkout_v2.rb [hunk 1]
```diff
@@ -109,6 +109,11 @@ def add_metadata(post, options)
 
       def add_payment_method(post, payment_method, options)
         post[:source] = {}
+        if options[:token_type] == 'id'
```
Comment (inai-shreyas) [inline lib/active_merchant/billing/gateways/checkout_v2.rb:112]: can you explain this `id` concept?

Comment (inai-shreyas) [reply lib/active_merchant/billing/gateways/checkout_v2.rb:112]: I feel it's a concept very specific to checkout maybe? let's generalize it.

Comment (inai-shreyas) [reply lib/active_merchant/billing/gateways/checkout_v2.rb:112]: maybe we can create something called PSPTokenizedCard and use type as `psp_token` to determine this?

File: test/remote/gateways/remote_checkout_v2_test.rb [hunk 2]
```diff
@@ -73,6 +86,12 @@ def test_successful_purchase_with_network_token
     assert_not_nil response.params['source']['payment_account_reference']
   end
 
+  def test_successful_purchase_with_source_token
+    response = @gateway.purchase(100, @source_id_payment_info, @source_id_payment_options)
+    assert_success response
+    assert_equal 'Succeeded', response.message
+  end
```
Comment (inai-shreyas) [inline test/remote/gateways/remote_checkout_v2_test.rb:93]: does this actually hit the PSP?

**Review by inai-shreyas — CHANGES_REQUESTED** [review]

File: test/remote/gateways/remote_checkout_v2_test.rb (hunk 2 above)
Comment (inai-harish) [reply test/remote/gateways/remote_checkout_v2_test.rb:93]: Yes it does @inai-shreyas

File: lib/active_merchant/billing/gateways/checkout_v2.rb (hunk 1 above)
Comment (inai-harish) [reply lib/active_merchant/billing/gateways/checkout_v2.rb]: I created a model `TokenizedCard` and kept the payment_data in it as a dict for now. And used the model for the check here

**Review by inai-shreyas — APPROVED** [review]

**Pull request merged**